x/market/internal/keeper/querier.go

```diff
@@ -35,30 +37,46 @@ func querySwap(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, s
 		return nil, types.ErrRecursiveSwap(types.DefaultCodespace, params.AskDenom)
 	}
 
-	swapCoin, spread, err := keeper.GetSwapCoin(ctx, params.OfferCoin, params.AskDenom, false)
+	swapCoin, spread, err := keeper.ComputeSwap(ctx, params.OfferCoin, params.AskDenom)
 	if err != nil {
 		return nil, sdk.ErrInternal(sdk.AppendMsgToErr("Failed to get swapped coin amount", err.Error()))
 	}
 
 	if spread.IsPositive() {
-		swapFeeAmt := spread.MulInt(swapCoin.Amount).TruncateInt()
+		swapFeeAmt := spread.Mul(swapCoin.Amount)
 		if swapFeeAmt.IsPositive() {
-			swapFee := sdk.NewCoin(swapCoin.Denom, swapFeeAmt)
+			swapFee := sdk.NewDecCoinFromDec(swapCoin.Denom, swapFeeAmt)
 			swapCoin = swapCoin.Sub(swapFee)
 		}
 	}
 
-	bz, err2 := codec.MarshalJSONIndent(keeper.cdc, swapCoin)
-	if err2 != nil {
-		return nil, sdk.ErrInternal(sdk.AppendMsgToErr("could not marshal result to JSON", err2.Error()))
+	retCoin, _ := swapCoin.TruncateDecimal()
+	bz, err := codec.MarshalJSONIndent(keeper.cdc, retCoin)
+	if err != nil {
+		return nil, sdk.ErrInternal(sdk.AppendMsgToErr("could not marshal result to JSON", err.Error()))
 	}
 
 	return bz, nil
 }
 
-func queryPrevDayIssuance(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
+func queryPoolInfo(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
```

Comment:
PoolInfo is poorly named; consider making more informative

Reply:
Changed